src/common/config/config.service.ts

```diff
@@ -12,6 +13,16 @@ export class ConfigService extends ConfigServiceSource<EnvironmentVariables> {
   }
 
   public get<T extends keyof EnvironmentVariables>(key: T): EnvironmentVariables[T] {
-    return super.get(key, { infer: true }) as EnvironmentVariables[T];
+    const value = super.get(key, { infer: true }) as EnvironmentVariables[T];
+
+    if (key !== 'DB_PASSWORD' || value !== undefined) {
+      return value;
+    }
+    const filePath = super.get('DB_PASSWORD_FILE', { infer: true });
```

Comment:
Is it expected to return the password when reading the `DB_PASSWORD_FILE` environment variable instead of returning file path?